src/docfx/restore/RestoreGit.cs

```diff
@@ -142,9 +143,13 @@ void AddWorkTrees()
                                         // clean existing work tree folder
                                         // it may be dirty caused by last failed restore action
                                         if (Directory.Exists(workTreePath))
+                                        {
+                                            // https://stackoverflow.com/questions/24265481/after-directory-delete-the-directory-exists-returning-true-sometimes
                                             Directory.Delete(workTreePath, true);
+                                            while (Directory.Exists(workTreePath))
+                                                Thread.Sleep(1);
```

Comment:
If we do need to wait, can there be a timeout, or a console log (like `RunInsideMutex`) to indicate that we are waiting for directory delete to complete.

Reply:
good suggestion for console log point, i will add one